app/views/search/index.html.erb

```diff
@@ -1 +1,52 @@
-<h1>hi</h1>
\ No newline at end of file
+<div class="container container-custom">
+  <h1 class="oompr-title">oompr</h1>
+  <div class="row">
+    <div class="col-md-6 offset-md-3">
+      <div class="listings-container">
+        <% @listings.each do |listing| %>
+          <div class="listing-item">
+            <div class="card card-custom mb-4">
+              <%= link_to listing_path(listing.id), class: "stretched-link text-decoration-none" do %>
+                <div class="card card-custom">
+                  <div id="carousel<%= listing.id %>" class="carousel slide" data-ride="carousel">
+                    <div class="carousel-inner">
+                      <div class="photos">
+                        <% listing.photos.each_with_index do |photo_url, index| %>
+                          <div class="carousel-item <%= 'active' if index == 0 %>">
+                            <%= image_tag photo_url, class: 'd-block w-100', alt: 'Listing Image' %>
+                          </div>
+                        <% end %>
+                      </div>
+                    </div>
+                    <a class="carousel-control-prev" href="#carousel<%= listing.id %>" role="button" data-slide="prev">
+                      <span class="carousel-control-prev-icon" aria-hidden="true"></span>
+                      <span class="sr-only">Previous</span>
+                    </a>
+                    <a class="carousel-control-next" href="#carousel<%= listing.id %>" role="button" data-slide="next">
+                      <span class="carousel-control-next-icon" aria-hidden="true"></span>
+                      <span class="sr-only">Next</span>
+                    </a>
+                  </div>
+                  <div class="card-body">
+                    <div class="price">
+                      <h5 class="card-title"><%= number_to_currency(listing.price) %></h5>
+                    </div>
+                    <div class="listing-info">
+                      <p class="card-text"><%= listing.address %></p>
+                      <p class="card-text">
+                        <i class="fas fa-bed"></i> <%= listing.bedrooms %>
+                        <i class="fas fa-bath"></i> <%= listing.bathrooms %>
+                        <i class="fas fa-expand-arrows-alt"></i> <%= listing.sqft %> m²
+                      </p>
+                    </div>
+                  </div>
+                </div>
+              <% end %>
+            </div>
+          </div>
+        <% end %>
+      </div>
+    </div>
+  </div>
+  <button class="btn btn-danger btn-lg reality-check-btn">Reality Check</button>
+</div>
```

Comment:
This page looks like a wild amount of work (I'm also biased because of how much I dislike typing out html) - good stuff!

Reply:
chatgpt didn't help me at all. :)